### PR TITLE
Support deleting loadbalancers when unit removed

### PR DESCRIPTION
**Pull request by dosaboy**
Closes-Bug: #1848610

### CODE REVIEW

File: lib/charms/layer/openstack.py [hunk 1]
```diff
@@ -345,6 +353,15 @@ def __init__(self, app_name, port, subnet, algorithm, fip_net,
...
         self._impl = self._get_impl()
 
+    def get_all(self):
+        lbs = []
+        for lb in self._impl.list_loadbalancers():
+            if lb['name'].startswith('openstack-integrator-{}'
+                                     .format(MODEL_SHORT_ID)):
+                lbs.append(lb)
+
+        return lbs
```
Comment (addyess) [inline lib/charms/layer/openstack.py]: here's an opportunity to use a nice list-generator
```suggestion
        return [
            lb for lb in self._impl.list_loadbalancers()
            if lb['name'].startswith('openstack-integrator-{}'.format(MODEL_SHORT_ID))
        ]
```

Comment (dosaboy) [reply lib/charms/layer/openstack.py]: I think you'll find that's a comprehension not a generator ;)

Comment (addyess) [reply lib/charms/layer/openstack.py]: sleepy-- yep

**Comment by johnsca** [comment]
Link to bug: https://bugs.launchpad.net/charm-openstack-integrator/+bug/1848610

File: lib/charms/layer/openstack.py [hunk 2]
```diff
@@ -140,10 +140,15 @@ def manage_loadbalancer(app_name, members):
 def cleanup():
     # note: we don't bother cleaning up the SG because it's a singleton
     # and can be reused in other / future deployments
-    for lb in LoadBalancer.get_all():
+    lbmanager = LoadBalancer('cleanup', None, None, None, None, None)
+    for lb in lbmanager.get_all():
         try:
-            lb.delete()
+            log("Deleting loadbalancer '{}'".format(lb['name']))
+            _openstack('loadbalancer', 'delete', '--cascade', lb['name'],
```
Comment (johnsca) [inline lib/charms/layer/openstack.py]: Does this work for Neutron-based LBs?

Comment (dosaboy) [reply lib/charms/layer/openstack.py]: Do we even support using neutron lbaas with k8s? Its deprecated in Openstack and replaced by Octavia. In any case, right now the charm does nothing so better to have it work for Octavia that neither?

**Pull request merged**